api/src/pcapi/routes/native/v1/authentication.py

```diff
@@ -129,11 +130,15 @@ def request_password_reset(body: RequestPasswordResetRequest) -> None:
 )
 def reset_password(body: ResetPasswordRequest) -> ResetPasswordResponse:
     check_password_strength("newPassword", body.new_password)
-
+    token = None
     try:
-        user = users_repo.get_user_with_valid_token(body.reset_password_token, [TokenType.RESET_PASSWORD])
+        token = token_utils.Token.load_and_check(body.reset_password_token, token_utils.TokenType.RESET_PASSWORD)
+        user = user_models.User.query.get(token.user_id)
     except users_exceptions.InvalidToken:
-        raise ApiErrors({"token": ["Le token de changement de mot de passe est invalide."]})
+        try:
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
**💣 chore:** ‏Un petit commentaire pour dire qu'on retire ça avec le ticket suivant ?